src/domain/discounts/discount-form/sections/conditions.tsx

```diff
@@ -1,22 +1,60 @@
 import React from "react"
+import NumberedItem from "../../../../components/molecules/numbered-item"
 import { useDiscountForm } from "../form/discount-form-context"
+import useConditionActions from "./useConditionActions"
 
 const Conditions = () => {
   const { conditions } = useDiscountForm()
+
+  const { getActions } = useConditionActions()
   return (
-    <div>
-      {Object.keys(conditions).map((key) => {
+    <div className="pt-6 flex flex-col gap-y-small">
+      {Object.keys(conditions).map((key, i) => {
         return (
           conditions[key] && (
-            <div>
-              <p>{key}</p>
-              <pre>{JSON.stringify(conditions[key], null, 2)}</pre>
-            </div>
+            <NumberedItem
+              index={i + 1}
+              title={getTitle(key)}
+              actions={getActions(key)}
+              description={
+                <ConditionSetting
+                  titles={conditions[key].map((c) => c.title)}
+                />
+              }
+            />
           )
         )
       })}
     </div>
   )
 }
 
+const ConditionSetting = ({ titles }) => {
+  const titleStrings = titles.reduce(
+    (acc, cur) =>
+      acc[1] + cur.length > 38 ? acc : [[...acc[0], cur], acc[1] + cur.length],
+    [[], 0]
+  )
```

Comment:
```suggestion
  const MAX_TITLES_LENGTH = 38
  
  const titleStrings = titles.reduce(
    (acc, cur) =>
      acc[1] + cur.length > MAX_TITLES_LENGTH ? acc : [[...acc[0], cur], acc[1] + cur.length],
    [[], 0]
  )
```

Maybe add something like this so we avoid magic numbers, and provide a bit of explanation for what the number 38 represents 😄